src/Microsoft.TemplateEngine.Cli/TemplateResolution/TemplateMatchInfoExtensions.cs

```diff
@@ -202,12 +202,52 @@ public static bool IsPartialMatchExceptContext(this ITemplateMatchInfo templateM
 
         public static bool HasNameMatchOrPartialMatch(this ITemplateMatchInfo templateMatchInfo)
         {
-            return templateMatchInfo.MatchDisposition.Any((x => (x.Location == MatchLocation.Name || x.Location == MatchLocation.ShortName) && (x.Kind == MatchKind.Exact || x.Kind == MatchKind.Partial)));
+            return templateMatchInfo.MatchDisposition.Any((x => (x.Location == MatchLocation.Name || x.Location == MatchLocation.ShortName || x.Location == MatchLocation.Classification) && (x.Kind == MatchKind.Exact || x.Kind == MatchKind.Partial)));
         }
 
         public static bool HasAnyMismatch(this ITemplateMatchInfo templateMatchInfo)
         {
             return templateMatchInfo.MatchDisposition.Any(m => m.Kind == MatchKind.Mismatch);
         }
+
+        public static bool HasClassificationMatchAndNameMismatch(this ITemplateMatchInfo templateMatchInfo)
+        {
+            if (templateMatchInfo.MatchDisposition.Count == 0)
+            {
+                return false;
+            }
+
+            bool hasNameMismatch = false;
+            bool hasClassificationMatch = false;
+
+            foreach (MatchInfo disposition in templateMatchInfo.MatchDisposition)
+            {
+                if (disposition.Location == MatchLocation.Name && disposition.Kind == MatchKind.Mismatch)
+                {
+                    hasNameMismatch = true;
+                }
+                else if (disposition.Location == MatchLocation.ShortName && disposition.Kind == MatchKind.Mismatch)
+                {
+                    hasNameMismatch = true;
+                }
```

Comment:
Current logic of name matching can return the following results:
- Name / Exact
- ShortName / Exact
- Name / Partial
- ShortName / Partial
- Name / Mismatch

I agree that concern makes sense in case both disposition on Name and ShortName could be added for same template but it can never happen. This method is not the only place relying on that.

Reply:
Oh, I see. In that case, let's keep it this way.

If it would cost the same time and effort, for similar future cases, my vote would be for implementing it in a "fool-proof" way despite the existing assumptions in the project. That may save some time when someone new is trying to understand and make changes to the code.